File: Palindrome-Number/Palindrome-Number.cpp

class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0)
            return false;
        int base=1;
        while(x/base>=10)
            base*=10;
        while(x)
        {
            int l=x/base;
            int r=x%10;
            if(l!=r)
                return false;
            x-=base*l;
            base/=100;
            x/=10;
        }
        return true;
    }
};


 

File: solutions/Palindrome-Number/Palindrome-Number.py


class Solution:
    # @return a boolean
    def isPalindrome(self, x):
        if x<0:
            return False
        y=x
        ans=0
        while x:
            temp=x%10
            ans=ans*10+temp
            x/=10
        if ans==y:
            return True
        else:
            return False
 